lib/ts/recipe/thirdparty/recipeImplementation.ts

```diff
@@ -13,106 +13,75 @@
  * under the License.
  */
 
-import STGeneralError from "../../error";
 import Querier from "../../querier";
 import { NormalisedAppInfo } from "../../types";
 import { appendQueryParamsToURL, getQueryParams, getSessionStorage, setSessionStorage } from "../../utils";
 import { UserType } from "../authRecipeWithEmailVerification/types";
 import { NormalisedInputType, RecipeInterface, StateObject } from "./types";
 import { RecipeFunctionOptions } from "../recipeModule/types";
 import { generateThirdPartyProviderState } from "./utils";
+import STGeneralError from "../../error";
 
 export default function getRecipeImplementation(recipeId: string, appInfo: NormalisedAppInfo): RecipeInterface {
     const querier = new Querier(recipeId, appInfo);
     return {
-        getOAuthState: function (): {
-            status: "OK";
-            state: StateObject | undefined;
-        } {
+        getStateAndOtherInfoFromStorage: function <CustomStateProperties>():
+            | (StateObject & CustomStateProperties)
+            | undefined {
             try {
-                const state = JSON.parse(getSessionStorage("supertokens-oauth-state"));
+                const stateFromStorage = getSessionStorage("supertokens-oauth-state");
 
-                if (state === null) {
-                    return {
-                        status: "OK",
-                        state: undefined,
-                    };
+                if (stateFromStorage === undefined) {
+                    return undefined;
                 }
 
+                const state = JSON.parse(stateFromStorage);
+
                 if (Date.now() > state.expiresAt) {
-                    return {
-                        status: "OK",
-                        state: undefined,
-                    };
+                    return undefined;
                 }
 
-                return {
-                    status: "OK",
-                    state,
-                };
+                return undefined;
```

Comment:
This function should only read from storage and return it as is. No checking of anything